File: microbenchmark/golang/utils/env_parser.go
package utils

import (
	"os"
	"strconv"
	"strings"
)

func ParseIntListFromEnv(envVarName string) []int {
	rawValue := os.Getenv(envVarName)
	rawParts := strings.Split(rawValue, ",")
	values := make([]int, len(rawParts))

	for i, part := range rawParts {
		value, err := strconv.Atoi(strings.TrimSpace(part))
		if err != nil {
			panic(err)
		}
		values[i] = value
	}

	return values
}

func ParseIntFromEnv(envVarName string) int {
	rawValue := os.Getenv(envVarName)
	value, err := strconv.Atoi(strings.TrimSpace(rawValue))
	if err != nil {
		panic(err)
	}
	return value
}

File: microbenchmark/python/src/utils/env_parser.py
import os


def ParseIntListFromEnv(envVarName: str) -> list[int]:
    rawValue: str = os.environ[envVarName]
    rawParts: list[str] = rawValue.split(",")

    values: list[int] = []
    for part in rawParts:
        values.append(int(part.strip()))

    return values


def ParseIntFromEnv(envVarName: str) -> int:
    rawValue: str = os.environ[envVarName]
    return int(rawValue.strip())
